middleware/instrument_test.go

```diff
@@ -0,0 +1,148 @@
+package middleware
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/prometheus/client_golang/prometheus/testutil"
+	"github.com/stretchr/testify/require"
+
+	"github.com/grafana/dskit/instrument"
+)
+
+func TestThroughputMetricHistogram(t *testing.T) {
+	tests := []struct {
+		name     string
+		sleep    bool
+		header   string
+		observed bool
+	}{
+		{"WithSleep", true, "unit=0, other_unit=2", true},
+		{"WithoutSleep", false, "unit=0, other_unit=2", false},
+		{"WithoutSleep", true, "", false},
+		{"WithoutSleep", false, "", false},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+
+			reg := prometheus.NewPedanticRegistry()
+			i := NewInstrument(reg)
+
+			wrap := i.Wrap(http.HandlerFunc(func(w http.ResponseWriter, _ *http.Request) {
+				if tt.sleep {
+					time.Sleep(i.RequestCutoff)
+				}
+				w.Header().Set("Server-Timing", tt.header)
+			}))
+
+			req := httptest.NewRequest("GET", "/", nil)
+			res := httptest.NewRecorder()
+
+			wrap.ServeHTTP(res, req)
+
+			output := ``
+			if tt.observed {
+				output = `
+				# HELP request_throughput_unit Server throughput running requests.
+				# TYPE request_throughput_unit histogram
+				request_throughput_unit_bucket{cutoff_ms="100",method="GET",route="other",le="1"} 1
+				request_throughput_unit_bucket{cutoff_ms="100",method="GET",route="other",le="5"} 1
+				request_throughput_unit_bucket{cutoff_ms="100",method="GET",route="other",le="10"} 1
+				request_throughput_unit_bucket{cutoff_ms="100",method="GET",route="other",le="+Inf"} 1
+				request_throughput_unit_sum{cutoff_ms="100",method="GET",route="other"} 0
+				request_throughput_unit_count{cutoff_ms="100",method="GET",route="other"} 1
+			`
+			}
+
+			require.NoError(t, testutil.GatherAndCompare(reg, strings.NewReader(output), "slow_request_throughput_"+i.ThroughputUnit))
+		})
+	}
+}
+
+func NewInstrument(registry *prometheus.Registry) Instrument {
```

Comment:
```suggestion
func newInstrument(registry *prometheus.Registry) Instrument {
```

this is only a test, so it doesn't make sense to export it